### DISCUSSION TITLE
Agent回放请求arex-storage接口/api/storage/record/query原理

### DISCUSSION THREAD

**Discussion by pony-qd** [arextest/arex-agent-java#593] [Q&A]
1)、/api/storage/record/query 接口设计原理有没有详细一点的文档说明
2)、同一个请求里面并发处理，可能会多次调用一个后端接口会出现概率性的请求不到mock数据

【遇到的问题】
部分使用接口请求mock数据显示无mock数据
使用postman模拟接口请求
1、第一次请求能正常返回数据
2、再次请求就返回空了。此时清理掉redis再请求就能正确返回

**Comment by lucas-myx** [comment]
/api/storage/record/query接口服务端实现逻辑可以参考[arex-storage](https://github.com/arextest/arex-storage)项目，
该接口是回放匹配接口，每次调用服务端会统计调用次数，举个例子：比如redis这个操作录制了2条数据，你回放调用了3次，那第3次就不会再返回数据，因为前两次已经匹配过了，第3次属于new call，也就是说这个接口会关联录制的数据和条数，不建议直接调用或通过postman调用
